### ISSUE TITLE
E02 leads to wonky temperatures

### ISSUE THREAD

**Issue by Skeletorjus** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#352]
After receiving an E02 I diacovered that the temperature was reported as zero. Ideally, this should be "unavailable" since the real value is unavailable for real, and in addition it won't mess up those pretty graphs... :)

Also, after restarting the pump the virtual temperature fell to around -100 before stabilizing at the correct value after about 20 minutes.

I've tried to capture how it went down in this screenshot:

![Screenshot_2022-07-22-23-33-24-55_c3a231c25ed346e59462e84656a70e50](https://user-images.githubusercontent.com/9031934/180571605-f2e15150-6056-435b-9253-26e64a69be79.jpg)

**Comment by visualapproach** [comment]
Okay. It sets the virtual temp to -99 at startup to indicate that it needs to be updated. It shouldn't show through mqtt though. I saw the same after the update but not the following reboots.
Can you confirm that it is like this when you make a soft reboot?

**Comment by visualapproach** [comment]
Disregard, I have the same behavior now so that will be fixed in a coming update

**Comment by Skeletorjus** [comment]
Thanks 👍
Forgot to mention that target temperature, both C and F, is reported as 100 after the restart.

![Screenshot_2022-07-24-18-19-34-43_40deb401b9ffe8e1df2f1cc5ba480b12](https://user-images.githubusercontent.com/9031934/180656495-f38047f6-b2a8-4dc9-89ec-512e76db75d4.jpg)

**Comment by visualapproach** [comment]
Thank you! The virtual temp should be fixed in latest dev branch. Need to take a look at target temp. Temperatures are reported both in C and F. In HA they will look the same because HA is converting everything to your preferred unit. Redundant sensors but it may be useful for other home automation systems.

**Comment by air101** [comment]
Is it possible to reset the pump from e02 via command line? or does it need a power supply restart?

**Comment by visualapproach** [comment]
It needs power cycle. Better to mitigate the cause of the error. Either your pipes or filter is clogged, or your impeller axle is worn. There is replacement kits to buy.

**Comment by visualapproach** [comment]
@Skeletorjus Problem should be solved in newest dev code

**Comment by Skeletorjus** [comment]
> @Skeletorjus Problem should be solved in newest dev code

Thanks a lot! Installed and will see how it goes. Seems promising so far.
A thing I just noticed is that when the automation to update the ambient temperature ran after the restart, I got -7 back from the sensor:
The value sent was 17.
Also saw the same yesterday after an E02.
![image](https://user-images.githubusercontent.com/9031934/180857937-032ae587-d08e-4753-a5f8-095765459bf3.png)
